src/p4est_io.h

```diff
@@ -57,9 +57,12 @@ SC_EXTERN_C_BEGIN;
 #define P4EST_NUM_USER_STRING_BYTES 48 /**< number of user string bytes */
 #define P4EST_NUM_FIELD_HEADER_BYTES (2 + P4EST_NUM_ARRAY_METADATA_BYTES + P4EST_NUM_USER_STRING_BYTES)
                                      /**< number of bytes of one field header */
-#define P4EST_FILE_MAX_GLOBAL_QUAD 9999999999999999 /**< maximal number of global quadrants */
-#define P4EST_FILE_MAX_BLOCK_SIZE 9999999999999 /**< maximal number of block bytes */
-#define P4EST_FILE_MAX_FIELD_ENTRY_SIZE 9999999999999 /**< maximal number of bytes per field entry */
+#define P4EST_FILE_MAX_GLOBAL_QUAD ((10000L * 10000L * 10000L * 10000L) - 1L)
```

Comment:
I'm still wondering. A factor of 10000UL is easy to confuse with 1000UL visually.

Shall we just revert this one and in the future define the maximum number of bytes, from which we calculate the maximum decimal number? That would be much easier on the eyes and brain.

Reply:
I agree and revert the commit that introduces this product representation.